servers/movie_writer/movie_writer_ogv.cpp

```diff
@@ -0,0 +1,441 @@
+		}
+	}
+
+	if (ogg_stream_pageout(&vo, &audio_page) > 0) {
+		return 1;
+	}
+
+	return 0;
+}
+
+int MovieWriterOGV::encode_video(const Ref<Image> &p_image) {
+	ogg_packet op;
+	if (ogg_stream_eos(&to)) {
+		return 0;
+	}
+
+	if (p_image != nullptr) {
+		PackedByteArray data = p_image->get_data();
+		rgb2yuv420(y, u, v, data.ptrw(), p_image->get_width(), p_image->get_height());
+
+		// We submit the buffer using the size of the picture region.
+		// libtheora will pad the picture region out to the full frame size for us,
+		// whether we pass in a full frame or not.
+		ycbcr[0].width = p_image->get_width();
+		ycbcr[0].height = p_image->get_height();
+		ycbcr[0].stride = p_image->get_width();
+		ycbcr[0].data = y;
+		ycbcr[1].width = p_image->get_width() / 2;
+		ycbcr[1].height = p_image->get_height() / 2;
+		ycbcr[1].stride = p_image->get_width() / 2;
+		ycbcr[1].data = u;
+		ycbcr[2].width = p_image->get_width() / 2;
+		ycbcr[2].height = p_image->get_height() / 2;
+		ycbcr[2].stride = p_image->get_width() / 2;
+		ycbcr[2].data = v;
+		th_encode_ycbcr_in(td, ycbcr);
+	}
+
+	int ret = 0;
+	do {
+		ret = th_encode_packetout(td, p_image == nullptr, &op);
+		if (ret > 0) {
+			ogg_stream_packetin(&to, &op);
+		}
+	} while (ret > 0);
+
+	if (ogg_stream_pageout(&to, &video_page) > 0) {
+		return 1;
+	}
+
+	return 0;
+}
+
+uint32_t MovieWriterOGV::get_audio_mix_rate() const {
+	return mix_rate;
+}
+
+AudioServer::SpeakerMode MovieWriterOGV::get_audio_speaker_mode() const {
+	return speaker_mode;
+}
+
+bool MovieWriterOGV::handles_file(const String &p_path) const {
+	return p_path.get_extension().to_lower() == "ogv";
+}
+
+void MovieWriterOGV::get_supported_extensions(List<String> *r_extensions) const {
+	r_extensions->push_back("ogv");
+}
+
+Error MovieWriterOGV::write_begin(const Size2i &p_movie_size, uint32_t p_fps, const String &p_base_path) {
+	base_path = p_base_path.get_basename();
+	if (base_path.is_relative_path()) {
+		base_path = "res://" + base_path;
+	}
+	base_path += ".ogv";
+
+	f = FileAccess::open(base_path, FileAccess::WRITE_READ);
+	ERR_FAIL_COND_V(f.is_null(), ERR_CANT_OPEN);
+
+	fps = p_fps;
+	speed = 4;
+
+	audio_ch = 2;
+	switch (speaker_mode) {
+		case AudioServer::SPEAKER_MODE_STEREO:
+			audio_ch = 2;
+			break;
+		case AudioServer::SPEAKER_SURROUND_31:
+			audio_ch = 4;
+			break;
+		case AudioServer::SPEAKER_SURROUND_51:
+			audio_ch = 6;
+			break;
+		case AudioServer::SPEAKER_SURROUND_71:
+			audio_ch = 8;
+			break;
+	}
+	audio_frames = mix_rate / fps;
+
+	// Set up Ogg output streams.
+	srand(time(NULL));
+	ogg_stream_init(&to, rand()); // Video.
+	ogg_stream_init(&vo, rand()); // Audio.
+
+	// Initialize Vorbis audio encoding.
+	vorbis_info_init(&vi);
+	int ret = 0;
+	if (audio_bitrate == 0) {
+		ret = vorbis_encode_init_vbr(&vi, audio_ch, mix_rate, audio_quality);
+	} else {
+		ret = vorbis_encode_init(&vi, audio_ch, mix_rate, -1, (int)(64870 * (ogg_int64_t)audio_bitrate >> 16), -1);
+	}
```

Comment:
`audio_bitrate` is always `0`, so the constant bitrate code (the `else` branch) is never run. We should remove it if there's no reason to use CBR over VBR.